ext/standard/streamsfuncs.c

```diff
@@ -616,93 +616,147 @@ PHP_FUNCTION(stream_get_wrappers)
 }
 /* }}} */
 
-/* {{{ stream_select related functions */
-static int stream_array_to_fd_set(zval *stream_array, fd_set *fds, php_socket_t *max_fd)
-{
-	zval *elem;
-	php_stream *stream;
-	int cnt = 0;
+/* {{{ fd_bigset related macros and typedefs to remove FD_SETSIZE limitation in stream_select */
+typedef struct {
+    char *fds_bits;
+    size_t size;
+} fd_bigset;
+#define FD_BIGSET_ENSURE_CAPACITY(fd, set) \
+    if (((fd) / 8 >= (set)->size)) { \
+        fd_bigset_double_size(set); \
+    }
+#define FD_BIGSET_ZERO(set, num_fds) do { \
+    (set)->size = (num_fds + 7) / 8; \
+    (set)->fds_bits = (char *)ecalloc((set)->size, sizeof(char)); \
+} while (0)
+#define FD_BIGSET_SET(fd, set) do { \
+	FD_BIGSET_ENSURE_CAPACITY(fd, set) \
+    if ((fd) / 8 < (set)->size) { \
```

Comment:
This condition is redundant with FD_BIGSET_ENSURE_CAPACITY()

Reply:
You are right. Thanks a lot for your detailed survey of the code :-) It's been years since I developed C.